internal/commands/permission_test.go

```diff
@@ -98,3 +103,79 @@ func TestCheckErrorWithInvalidDebugInformation(t *testing.T) {
 	require.NotNil(t, err)
 	require.ErrorContains(t, err, "unknown field: invalid")
 }
+
+func TestLookupResourcesCommand(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	srv := zedtesting.NewTestServer(ctx, t)
+	go func() {
+		require.NoError(t, srv.Run(ctx))
+	}()
+	conn, err := srv.GRPCDialContext(ctx)
+	require.NoError(t, err)
+
+	originalClient := client.NewClient
+	defer func() {
+		client.NewClient = originalClient
+	}()
+
+	client.NewClient = zedtesting.ClientFromConn(conn)
+
+	c, err := zedtesting.ClientFromConn(conn)(nil)
+	require.NoError(t, err)
+
+	_, err = c.WriteSchema(ctx, &v1.WriteSchemaRequest{Schema: testSchema})
+	require.NoError(t, err)
+
+	var updates []*v1.RelationshipUpdate
+	for i := 0; i < 10; i++ {
+		updates = append(updates, &v1.RelationshipUpdate{
+			Operation:    v1.RelationshipUpdate_OPERATION_TOUCH,
+			Relationship: tuple.ParseRel(fmt.Sprintf("test/resource:%d#reader@test/user:1", i)),
+		})
+	}
+
+	_, err = c.WriteRelationships(ctx, &v1.WriteRelationshipsRequest{Updates: updates})
+	require.NoError(t, err)
+
+	previous := console.Println
+	defer func() {
+		console.Println = previous
+	}()
+	var count int
+	console.Println = func(values ...any) {
+		count += len(values)
+	}
+
+	// test no page size, server computes returns all resources in one go
+	cmd := testLookupResourcesCommand(t, 0)
+	err = lookupResourcesCmdFunc(cmd, []string{"test/resource", "read", "test/user:1"})
+	require.NoError(t, err)
+	require.Equal(t, 10, count)
+
+	// use page size same as number of elements
+	count = 0
+	cmd = testLookupResourcesCommand(t, 10)
+	err = lookupResourcesCmdFunc(cmd, []string{"test/resource", "read", "test/user:1"})
+	require.NoError(t, err)
+	require.Equal(t, 10, count)
+
+	// use odd page size
+	count = 0
+	cmd = testLookupResourcesCommand(t, 3)
+	err = lookupResourcesCmdFunc(cmd, []string{"test/resource", "read", "test/user:1"})
```

Comment:
It would be nice if we could validate that the parameter was sent to the server and that the responses are limiting the page size, but I'm unfamiliar with the test server code and whether or not that is possible right now.

Reply:
That's a good point. I added some plumbing to be able to test this: https://github.com/authzed/zed/pull/395/commits/6a6c01a1881b6210295318bdeaa6ba97e4125b6b

Reply:
Nice!